### PR TITLE
Major UI update and start of numerical analysis functionality

### PR DESCRIPTION
**Pull request by user17359**

### CODE REVIEW

File: lib/overview.dart [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+              ),
+              floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
+              floatingActionButton: OutlinedButton.icon(
+                onPressed: () async {
+                  Uri url = Uri.parse(snapshot.data![index]["wikipedia_page"]);
+                  if (await canLaunchUrl(url)) {
+                    await launchUrl(url, mode: launchMode);
+                  } else {
+                    throw 'Could not launch $url';
+                  }
```
Comment (HulekJakub) [inline lib/overview.dart:106]: I see this pattern very often. Wouldn't it be better for a utility class to contain this?

File: lib/numerical_analysis/find_peaks.dart [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          for(int j = i + 1; j < data.length && !hit; j++)
+          {
+            if(dataConverted[j].y >= dataConverted[i].y || j == (data.length - 1)){
+              hit = true;
+            }
+            else {
+              if (dataConverted[j].y < referenceY) {
+                referenceY = dataConverted[j].y;
+              }
+            }
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:26]: Maybe `referenceY = min(referenceY, dataConverted[j].);`

File: lib/numerical_analysis/find_peaks.dart [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          for(int j = i - 1; j >= 0 && !hit; j--)
+          {
+            if(dataConverted[j].y >= dataConverted[i].y || j == 0){
+              hit = true;
+            }
+            else {
+              if (dataConverted[j].y < referenceY) {
+                referenceY = dataConverted[j].y;
+              }
+            }
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:46]: Maybe `referenceY = min(referenceY, dataConverted[j].);`

File: lib/numerical_analysis/find_peaks.dart [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          for(int j = i + 1; j < data.length && !hit; j++)
+          {
+            if(dataConverted[j].y >= dataConverted[i].y || j == (data.length - 1)){
+              hit = true;
+            }
+            else {
+              if (dataConverted[j].y < referenceY) {
+                referenceY = dataConverted[j].y;
+              }
+            }
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:86]: ^^

File: lib/numerical_analysis/find_peaks.dart [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          }
+          rightMinimum = referenceY;
+
+          //choosing bigger value as per algorithm
+          if(rightMinimum > leftMinimum){
+            referenceY = rightMinimum;
+          }
+          else{
+            referenceY = leftMinimum;
+          }
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:96]: Wouldn't `referenceY = max(rightMinimum, leftMinimum)`  be better? max function is available in dart:math

File: lib/numerical_analysis/find_peaks.dart [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            }
+          }
+          leftMinimum = referenceY;
+          hit = false;
+          //check to the right
+          for(int j = i + 1; j < data.length && !hit; j++)
+          {
+            if(dataConverted[j].y >= dataConverted[i].y || j == (data.length - 1)){
+              hit = true;
+            }
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:81]: I think 
`if(dataConverted[j].y >= dataConverted[i].y){
      hit = true;
 }`
is enough, loop will end either way, no need to set hit = true

File: lib/numerical_analysis/find_peaks.dart [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+          for(int j = i - 1; j >= 0 && !hit; j--)
+          {
+            if(dataConverted[j].y >= dataConverted[i].y || j == 0){
+              hit = true;
+            }
+            else {
+              if (dataConverted[j].y < referenceY) {
+                referenceY = dataConverted[j].y;
+              }
+            }
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:72]: Maybe `referenceY = min(referenceY, dataConverted[j].);`

File: lib/numerical_analysis/find_peaks.dart [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+  List<LinearData> dataConverted = [];
+  List<LinearData> peaks = [];
+  for (double key in data.keys) {
+    dataConverted.add(LinearData(key, data[key]!));
+  }
+
+  for(int i = 0; i < data.length; i++) {
+      //edge cases
+      if(i == 0){
```
Comment (HulekJakub) [inline lib/numerical_analysis/find_peaks.dart:13]: Could we add one LinearData at the start and one at the end, both with value of 0 and just get rid of those ifs? Just one `for(int i = 1; i < data.length - ;i++){[...]} `

**Review by HulekJakub — CHANGES_REQUESTED** [review]
Please read through my review. It would be beneficial to make a separate directory for UI pages

**Comment by HulekJakub** [comment]
A small question. Shouldn't peak analysis have a separate branch?

**Comment by user17359** [comment]
>A small question. Shouldn't peak analysis have a separate branch?
It does, but merge was needed to create UI for that part of app.

**Review by HulekJakub — APPROVED** [review]
Okay

**Pull request merged**